packages/errors/src/react/error-page.tsx

```diff
@@ -1,116 +1,200 @@
-    media(
-      '(min-width: 769px)',
-      stylesheet['@media (min-width: 769px)']['.psds-error-page__text']
-    ),
-  code: (themeName: ValueOf<typeof themeNames>) =>
-    css(
-      stylesheet['.psds-error-page__code'],
-      stylesheet[`.psds-error-page__code.psds-theme--${themeName}`]
-    ),
-  search: (themeName: ValueOf<typeof themeNames>) => {
-    const label = 'psds-error-page__search'
+interface ErrorPageStatics {
+  Actions: typeof Actions
+  Caption: typeof Caption
+  ErrorCode: typeof ErrorCode
+  Heading: typeof Heading
+  Illustration: typeof Illustration
+  sizes: typeof sizes
+}
 
-    return compose(
-      css(stylesheet[`.${label}`]),
-      css(stylesheet[`.${label}.psds-theme--${themeName}`])
-    )
-  },
-  searchIcon: (themeName: ValueOf<typeof themeNames>) =>
-    css(
-      stylesheet['.psds-error-page__search__icon'],
-      stylesheet[`.psds-error-page__search__icon.psds-theme--${themeName}`]
+interface ErrorPageComponent
+  extends RefForwardingComponent<
+    ErrorPageProps,
+    HTMLDivElement,
+    ErrorPageStatics
+  > {}
+
+type StyleFn = (
+  props: unknown,
+  ctx: ContextValue,
+  opts?: unknown
+) => StyleAttribute
+
+const renderSmallIfElementLessThan = 450
+
+const styles = {
+  errors: (ctx: ContextValue, hasRenderedOnce: boolean) =>
+    compose(
+      combineClasses('.psds-error-page', ctx),
+      !hasRenderedOnce && stylesheet['.psds-error-page--hidden']
     ),
-  searchInput: (themeName: ValueOf<typeof themeNames>) => {
-    const label = 'psds-error-page__search__input'
+  actions: (ctx: ContextValue) =>
+    combineClasses('.psds-error-page__actions', ctx),
+  caption: (ctx: ContextValue) =>
+    combineClasses('.psds-error-page__caption', ctx),
+  errorCode: (ctx: ContextValue) =>
+    combineClasses('.psds-error-page__error-code', ctx),
+  heading: (ctx: ContextValue) =>
+    combineClasses('.psds-error-page__heading', ctx),
+  illustration: (ctx: ContextValue) =>
+    combineClasses('.psds-error-page__illustration', ctx)
+}
+
+const combineClasses = (className: string, { size, themeName }: ContextValue) =>
+  css(
+    stylesheet[className],
+    stylesheet[className + themeClasses[themeName as string]],
+    stylesheet[className + sizeClasses[size as string]]
+  )
+
+const ErrorPage = forwardRef<HTMLDivElement, ErrorPageProps>(
+  (props, forwardedRef) => {
+    const {
+      actions,
+      caption,
+      heading,
+      illustration,
+      errorCode,
+      size: sizeOverride,
+      ...rest
+    } = props
+
+    const ref = useRef<HTMLDivElement>()
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    useImperativeHandle(forwardedRef, () => ref.current as HTMLDivElement)
 
-    return compose(
-      css(stylesheet[`.${label}`]),
-      css(stylesheet[`.${label}.psds-theme--${themeName}`])
+    const themeName = useTheme()
+
+    const [hasRenderedOnce, setHasRenderedOnce] = useState(false)
+    const { width } = useResizeObserver(ref)
```

Comment:
Maybe I should shy away from it, but a resize observe for some reason feels like a potentially-heavy bit of computation. And here, imo, for a pretty low-value payoff.  (same with emptystate). Feels like we could just leave it to apps to decide: Do you want a small or regular-sized error page?  Set the prop, be done. We're not going to do component dimension observation so we can do minor size bumps. Hehe, anyway, not sure that'll change the outcome here, but in case anyone was thinking the same thing, I thought I'd say it. :)

Reply:
I kinda like it honestly. This was copied over from empty state. In both case they are usually one of only a few nodes and active modules running on the window. I feel in such situations it's not to expensive or likely to cause jank.